core/app/models/spree/order.rb

```diff
@@ -524,9 +525,15 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
-      adjustments.shipping.destroy_all
-      shipments.destroy_all
-      self.shipments = Spree::Stock::Coordinator.new(self).shipments
+      if completed?
+        raise CannotRebuildShipments.new(Spree.t(:cannot_rebuild_shipments_order_completed))
+      elsif shipments.any? { |s| !s.pending? }
```

Comment:
Unreturned expedited exchanges, where the shipments are already shipped and the order is attempting to complete. It should not call this method, but if it does, then at least the shipments don't get destroyed.